src/main/java/world/bentobox/border/Border.java

```diff
@@ -1,36 +1,120 @@
 package world.bentobox.border;
 
+import com.github.yannicklamprecht.worldborder.api.BorderAPI;
+import com.github.yannicklamprecht.worldborder.api.WorldBorderApi;
 import org.bukkit.Bukkit;
+import org.bukkit.Location;
+import org.bukkit.entity.Player;
 import org.bukkit.plugin.Plugin;
-
+import org.bukkit.scheduler.BukkitRunnable;
+import org.eclipse.jdt.annotation.NonNull;
+import org.eclipse.jdt.annotation.Nullable;
 import world.bentobox.bentobox.api.addons.Addon;
+import world.bentobox.bentobox.database.Database;
+import world.bentobox.bentobox.database.objects.Island;
+import world.bentobox.border.commands.IslandBorderCommand;
 import world.bentobox.border.listeners.PlayerListener;
 
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+import java.util.UUID;
+import java.util.function.Consumer;
+
 public final class Border extends Addon {
-    
-    @Override
-    public void onLoad() {
-        // Nothing to do
+
+    private WorldBorderApi worldBorderApi;
+    private Database<BorderData> handler;
+    private Map<UUID, BorderData> borderCache;
+
+    public void getLevelsData(@NonNull UUID targetPlayer, Consumer<BorderData> callback) {
+        BorderData borderData = borderCache.get(targetPlayer);
+        if (borderData != null) {
+            callback.accept(borderData);
+            return;
+        }
+        new BukkitRunnable() {
+            @Override
+            public void run() {
+                String uniqueId = targetPlayer.toString();
+                BorderData data = Optional.ofNullable(handler.loadObject(uniqueId))
+                        .orElseGet(() -> new BorderData(uniqueId));
+                new BukkitRunnable() {
+                    @Override
+                    public void run() {
+                        borderCache.put(targetPlayer, data);
+                        callback.accept(data);
+                    }
+                }.runTask(getPlugin());
+            }
+        }.runTaskAsynchronously(getPlugin());
+    }
+
+    public void uncachePlayer(@Nullable UUID uniqueId) {
+        BorderData data = borderCache.remove(uniqueId);
+        if (data == null) {
+            return;
+        }
+        new BukkitRunnable() {
+
+            @Override
+            public void run() {
+                handler.saveObject(data);
+            }
+        }.runTaskAsynchronously(getPlugin());
```

Comment:
SaveObject is async so this wouldn't be required. Doesn't harm though.

Reply:
Nah, I'll remove it as it's a waste of resources